### PR TITLE
Move to Github Actions and python version upgrade

### PR DESCRIPTION
**Pull request by shiplu**
I wanted to fix the Travis build errors and support Python 3.7. That lead me to many other changes. Of course, you can frown.
I used Google BigQuery to check the [PyPI download statistics of zerorpc](https://docs.google.com/spreadsheets/d/1lcPGd0BWrmeSjccXRLuk4tRZR2I-HZp1kM39TFzKhbA/edit?usp=sharing) before dropping support of old python versions.

- Dropped 2.6, 3.4, 3.5 support
- Introduced 3.6, 3.7, 3.8 and 3.9
- Use "bionic" on for the travis build
- Pass 'async' as ** dict item in the test so it's compatible with newer
versions of python
- Use pyzmq 16.0.4+ as the old versions cannot be built in python 3.7
- Support modern repr of BaseException
  BaseException doesn't add trailing comma from version 3.7.0.
  Hence the test case without ',' is added as well.
  See also: https://bugs.python.org/issue30399

### CODE REVIEW

**Comment by shiplu** [comment]
Not sure why the Travis checks are not triggered. It triggered well in [my fork](https://github.com/shiplu/zerorpc-python/pull/3)

**Comment by gabrielgrant** [comment]
Thanks for this @shiplu 

+1 on dropping 2.6 support, it's been EOLed for years now. Rest of the changes look good to me, but haven't been through this with a fine-toothed comb. 

@bombela any idea why Travis isn't getting triggered? Is that something you can do manually?

**Comment by shiplu** [comment]
Maybe travis is dropping support. @bombela A small nudge. Would you have time to review this?

**Comment by shiplu** [comment]
I have removed travis support and add github actions instead. The tests are running in [my own PR](https://github.com/shiplu/zerorpc-python/pull/3). Still now sure why it's not being triggered in this PR.

File: tests/test_heartbeat.py [hunk 1]
```diff
@@ -162,7 +161,7 @@ def test_heartbeat_can_open_channel_client_close():
     print('SERVER LOST CLIENT :)')
     server_hbchan.close()
     server.close()
-
+    client.close()
```
Comment (shiplu) [inline tests/test_heartbeat.py:164]: This test `test_heartbeat_can_open_channel_client_close` fails in _Python 3.9_ if the client is closed before the server heartbeat channel is closed. Looking at the other tests, it seemed to me that `client.close()` should be called _after_ `server.close()`.

Comment (bombela) [reply tests/test_heartbeat.py:164]: This defeats the purpose of this test. The point is to have the client go away. And the server to detect this by the loss of heartbeats.

Comment (shiplu) [reply tests/test_heartbeat.py:164]: I see. Then I'll change it back. What I found that the test fails on [python 3.9](https://github.com/shiplu/zerorpc-python/actions/runs/1013103813) if the client is closed before. Any idea why it fails on Python 3.9?

Comment (bombela) [reply tests/test_heartbeat.py:164]: The test itself is successful. You can clearly see the message "CLOSE CLIENT SOCKET!!!". Followed by "SERVER LOST CLIENT :)". This is the expected behavior.

Somehow the test is marked as failed. Maybe the assert_raises is not doing its job.

Comment (shiplu) [reply tests/test_heartbeat.py]: Thanks for the confirmation. I'll see if I can fix it!

Comment (shiplu) [reply tests/test_heartbeat.py:164]: It seems the issue is on nosetests. I run many times these tests on python 3.9. It never fails. Only on github it fails. Then I found one of the tests `` passing with `ok` status.

```
tests.test_buffered_channel.test_congestion_control_server_pushing ... Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 906, in gevent._gevent_cgree nlet.Greenlet.run
  File "/__w/zerorpc-python/zerorpc-python/tests/test_buffered_channel.py", line 429, in server_do
    server_bufchan.emit('coucou', x, timeout=0)  # will fail when x == 1
  File "/usr/local/lib/python3.9/unittest/case.py", line 226, in __exit__
    self._raiseFailure("{} not raised".format(exc_name))
  File "/usr/local/lib/python3.9/unittest/case.py", line 163, in _raiseFailure
    raise self.test_case.failureException(msg)
AssertionError: TimeoutExpired not raised
2021-07-13T21:28:53Z <Greenlet at 0x7efdea2d5260: server_do> failed with AssertionError

ok
```
`AssertionError` and `ok` status together shouldn't be right.

The errors can be seen in my [github action build log](https://github.com/shiplu/zerorpc-python/runs/3061176173?check_suite_focus=true)

So, I think for now we drop supporting python 3.9 (removing the entry from ci.yml and not adding in setup.py). Then probably we can migrate from nosetests to pytest which will enable us to support 3.9 in near future!

**Comment by shiplu** [comment]
@bombela Any comments?

**Comment by chombourger** [comment]
Not sure what @bombela thinks but it would IMO help if we break this PR into smaller ones (e.g. raise a separate PR to fix Flake8 errors)

**Comment by bombela** [comment]
I agree, one PR per topic. In order of reverse dependency of course.

Dropping python 2.6 might be reasonable. For other versions of python 3, I am not so sure. I hate it when things suddenly stop working after an upgrade because people ditch backward compatibility.

**Comment by shiplu** [comment]
Closing this PR. I woudln't have time to work on it. Feel free to take ownership.

**Pull request closed without merging**